MinecraftClient/ChatBots/MineCube.cs

```diff
@@ -0,0 +1,517 @@
+﻿using System;
+using System.Collections.Generic;
+using MinecraftClient.Mapping;
+using MinecraftClient.Inventory;
+using System.Threading;
+
+namespace MinecraftClient.ChatBots
+{
+    /// <summary>
+    /// A row of blocks that will be mined
+    /// </summary>
+    public class Row
+    {
+        private List<Location> blocksInRow;
+
+        public List<Location> BlocksToMine
+        {
+            get { return blocksInRow; }
+        }
+
+        /// <summary>
+        /// Initialize a row of blocks
+        /// </summary>
+        /// <param name="bIL"> Enter a list of blocks </param>
+        public Row(List<Location> bIL = null)
+        {
+            blocksInRow = bIL ?? new List<Location>();
+        }
+    }
+
+    /// <summary>
+    /// Several rows are summarized in a layer
+    /// </summary>
+    public class Layer
+    {
+        private List<Row> rowsToMine;
+
+        public List<Row> RowsToMine
+        {
+            get { return rowsToMine; }
+        }
+
+        /// <summary>
+        /// Add a new row to this layer
+        /// </summary>
+        /// <param name="givenRow"> enter a row that should be added </param>
+        /// <returns> Index of the last row </returns>
+        public int addRow(Row givenRow = null)
+        {
+            rowsToMine.Add(givenRow ?? new Row());
+            return rowsToMine.Count - 1;
+        }
+
+        /// <summary>
+        /// Initialize a layer
+        /// </summary>
+        /// <param name="rTM"> Enter a list of rows </param>
+        public Layer(List<Row> rTM = null)
+        {
+            rowsToMine = rTM ?? new List<Row>();
+        }
+    }
+
+    /// <summary>
+    /// Several layers result in a cube
+    /// </summary>
+    public class Cube
+    {
+        private List<Layer> layersToMine;
+
+        public List<Layer> LayersToMine
+        {
+            get { return layersToMine; }
+        }
+
+        /// <summary>
+        /// Add a new layer to the cube
+        /// </summary>
+        /// <param name="givenLayer"> Enter a layer that should be added </param>
+        /// <returns> Index of the last layer </returns>
+        public int addLayer(Layer givenLayer = null)
+        {
+            layersToMine.Add(givenLayer ?? new Layer());
+            return layersToMine.Count - 1;
+        }
+
+        /// <summary>
+        /// Initialize a cube
+        /// </summary>
+        /// <param name="lTM"> Enter a list of layers </param>
+        public Cube(List<Layer> lTM = null)
+        {
+            layersToMine = lTM ?? new List<Layer>();
+        }
+    }
+
+    class MineCube : ChatBot
+    {
+        public override void Initialize()
+        {
+            if (!GetTerrainEnabled())
+            {
+                LogToConsole(Translations.Get("extra.terrainandmovement_required"));
+                UnLoadBot(this);
```

Comment:
Use this instead of `UnLoadBot(this)`
https://github.com/ORelio/Minecraft-Console-Client/blob/982f358bf37b69ac1d99887887d71fe86226a2fb/MinecraftClient/ChatBot.cs#L836-L839

Also remember to `return` after calling `UnloadBot()` or the `RegisterChatBOtCommand()` below would be called.
```suggestion
                UnloadBot();
                return;
```